audio/out/ao_aaudio.c

```diff
@@ -0,0 +1,454 @@
+        p->AAudioStreamBuilder_delete(p->builder);
+        p->builder = NULL;
+        return -1;
+    }
+
+    MP_ERR(ao, "Failed to create stream builder: %s\n",
+           p->AAudio_convertResultToText(result));
+    return -1;
+}
+
+static void start(struct ao *ao)
+{
+    struct priv *p = ao->priv;
+
+    aaudio_result_t result = AAUDIO_OK;
+    if (!p->stream) {
+        if ((result = p->AAudioStreamBuilder_openStream(p->builder, &p->stream)) >= 0)
+            ao->device_buffer = p->AAudioStream_getBufferCapacityInFrames(p->stream);
+    }
+
+    if (result >= 0) {
+        aaudio_stream_state_t next;
+
+        if ((result = p->AAudioStream_requestStart(p->stream)) >= 0)
+            result = p->AAudioStream_waitForStateChange(
+                p->stream, AAUDIO_STREAM_STATE_STARTING, &next, INT64_MAX);
+    }
+
+    if (result < 0)
+        MP_ERR(ao, "Failed to start stream: %s\n", p->AAudio_convertResultToText(result));
+}
+
+static bool set_pause(struct ao *ao, bool paused)
+{
+    struct priv *p = ao->priv;
+
+    aaudio_result_t result;
+    aaudio_stream_state_t state, next;
+
+    if (paused) {
+        result = p->AAudioStream_requestPause(p->stream);
+        state = AAUDIO_STREAM_STATE_PAUSING;
+    } else {
+        result = p->AAudioStream_requestStart(p->stream);
+        state = AAUDIO_STREAM_STATE_STARTING;
+    }
+
+    if (result >= 0) {
+        result = p->AAudioStream_waitForStateChange(p->stream, state, &next, INT64_MAX);
+        return true;
+    }
+
+    MP_ERR(ao, "Failed to pause stream: %s\n", p->AAudio_convertResultToText(result));
+    return false;
+}
+
+static void reset(struct ao *ao)
+{
+    struct priv *p = ao->priv;
+
+    if (p->stream) {
+        p->AAudioStream_close(p->stream);
+        p->stream = NULL;
+    }
+}
+
+// clang-format off
+#define OPT_BASE_STRUCT struct priv
+
+const struct ao_driver audio_out_aaudio = {
+    .description = "AAudio audio output",
+    .name = "aaudio",
+    .init = init,
+    .uninit = uninit,
+    .start = start,
+    .reset = reset,
+    .set_pause = set_pause,
+
+    .priv_size = sizeof(struct priv),
+    .priv_defaults = &(const struct priv) {
+        .device_id = AAUDIO_UNSPECIFIED,
+        .buffer_capacity = AAUDIO_UNSPECIFIED,
+        .performance_mode = AAUDIO_PERFORMANCE_MODE_NONE,
+        .stream = NULL,
+        .lib_handle = NULL
+    },
+    .options_prefix = "aaudio",
+    .options =
+        (const struct m_option[]){
+            {"device-id", OPT_CHOICE(device_id, {"auto", AAUDIO_UNSPECIFIED}),
+             M_RANGE(1, 96000)},
+            {"buffer-capacity", OPT_CHOICE(buffer_capacity, {"auto", AAUDIO_UNSPECIFIED}),
+             M_RANGE(1, 96000)},
+            {"performance-mode",
+             OPT_CHOICE(performance_mode, {"none", AAUDIO_PERFORMANCE_MODE_NONE},
+                        {"low-latency", AAUDIO_PERFORMANCE_MODE_LOW_LATENCY},
+                        {"power-saving", AAUDIO_PERFORMANCE_MODE_POWER_SAVING})},
+            {0}},
```

Comment:
you let clang-format touch this code, right? :wink: 
```suggestion
    .options = (const struct m_option[]) {
        {"device-id", OPT_CHOICE(device_id, {"auto", AAUDIO_UNSPECIFIED}),
            M_RANGE(1, 96000)},
        {"buffer-capacity", OPT_CHOICE(buffer_capacity, {"auto", AAUDIO_UNSPECIFIED}),
             M_RANGE(1, 96000)},
        {"performance-mode",
             OPT_CHOICE(performance_mode,
                {"none", AAUDIO_PERFORMANCE_MODE_NONE},
                {"low-latency", AAUDIO_PERFORMANCE_MODE_LOW_LATENCY},
                {"power-saving", AAUDIO_PERFORMANCE_MODE_POWER_SAVING})},
        {0}
    },
```